types/react/v15/index.d.ts

```diff
@@ -342,13 +342,17 @@ declare namespace React {
     // Class Interfaces
     // ----------------------------------------------------------------------
 
+    type NonNullable<T> = T & {};
```

Comment:
What's the purpose of this?

Reply:
`StatelessComponent` returns `ReactElement | null`. In order to access the `props` property, we have to assert the element is non-null.

This type will be included in the standard lib starting TypeScript 2.8.